package fetch.cfg.device;

message NIDAQZPiezo
{ optional double um2v      = 1 [default=0.025];        // V/um - approximate micron to volts converstion (for control output). e.g. 1 um = 0.025 V = 25 mV
  optional double v_lim_max = 5 [default=12.0];         // V    - Maximum permissible value
  optional double v_lim_min = 6 [default=-3.0];         // V    - Minimum permissible value
  optional double v_offset  = 7 [default=0.0];          // V    - Voltage corresponding to the spatial zero point
  optional string channel   = 8 [default="/Dev1/ao1"];  // DAQ terminal: should be connected to command input on galvo controller
}

message SimulatedZPiezo
{ optional double dummy = 1 [default=0];
}

message vDAQZPiezo
{ optional double um2v      = 1 [default=0.025];        // V/um - approximate micron to volts converstion (for control output). e.g. 1 um = 0.025 V = 25 mV
  optional double v_lim_max = 5 [default=12.0];         // V    - Maximum permissible value
  optional double v_lim_min = 6 [default=-3.0];         // V    - Minimum permissible value
  optional double v_offset  = 7 [default=0.0];          // V    - Voltage corresponding to the spatial zero point
  optional int32  channel   = 8 [default=4];            // DAQ terminal: should be connected to command input on galvo controller
}

message ZPiezo
{
  enum ZPiezoType
  {
    Simulated = 0;
    NIDAQ     = 1;
    vDAQ      = 2;
  }  
  optional ZPiezoType      kind      = 1 [default=vDAQ];
  optional NIDAQZPiezo     nidaq     = 2;  
  optional SimulatedZPiezo simulated = 3;
  optional vDAQZPiezo      vdaq      = 4;
  
  optional double um_max    = 5 [default=100.0];        // um   - end of travel for a scan
  optional double um_min    = 6 [default=0.0];          // um   - beg of travel for a scan
  optional double um_step   = 7 [default=1.00];         // um   - step to take during a vertical scan
}